huaweicloud/services/acceptance/ram/data_source_huaweicloud_ram_shared_principals_test.go

```diff
@@ -0,0 +1,78 @@
+package ram
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+// Before executing this use case, please create several pieces of data first.
+func TestAccDatasourceRAMSharedPrincipals_basic(t *testing.T) {
+	rName := "data.huaweicloud_ram_shared_principals.test"
+	principalTestname := "data.huaweicloud_ram_shared_principals.principal_filter"
+	resourceURNTestName := "data.huaweicloud_ram_shared_principals.resource_urn_filter"
+	dc := acceptance.InitDataSourceCheck(rName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckRAMSharedPrincipals(t)
+			acceptance.TestAccPreCheckRAM(t)
```

Comment:
Please write a separate preCheck. There are two unnecessary parameters in the current preCheck.